feat: add consistent timestamps to (almost) all items in the player ui

Aeolun opened the pull request:
This also tries to make the autoscroll functionality a bit more consistent, where all items are always shown in the list, but items which have not yet occurred will be partially transparent until they happen.

Due to that change, autoscroll behavior which previously always went all the way to the bottom of a list didn't make sense anymore, so now it scrolls to the current item.

It also fixes an issue where GraphQL requests do not have a 'duration' specified, and the waterfall graph doesn't correctly display, though to do this it now assumes a default duration of 200ms for all GraphQL requests.

## Notes

- This isn't yet implemented for the store component since we do not have that integrated into our application.
- I converted a few components to TS where that was relevant.

## Screenshots

Console
![image](https://user-images.githubusercontent.com/1116482/184809922-80565b23-b747-439d-b720-147bfeb78219.png)

Network
![image](https://user-images.githubusercontent.com/1116482/184809986-3ccee92e-b3f4-495c-93ea-e38ef0d47ac1.png)

Exceptions
![image](https://user-images.githubusercontent.com/1116482/184810529-2af35290-31c4-4a88-ba14-5317e3010ff1.png)

Events
![image](https://user-images.githubusercontent.com/1116482/184810732-966c9962-5ea5-4393-a105-b7bdf13c8f16.png)

sylenien commented:
Hi!
Thanks for the contribution. Can you please change your target branch to `dev` and rebase on it as well plus remove package lock file?

Side note: @types/luxon is adding some additional type errors in the older code & we try to use `moment` now.

sylenien commented:
Would be great if you could also ammend commit name to feat**(ui)**: add consistent timestamps to (almost) all items in the player ui

sylenien commented:
Updating the base branch of this feature branch to local dev (and updating local one to match ours) will fix the conflicts and cut the additional commits

Aeolun commented:
@sylenien Ok, everything is rebased and cleaned up. I've removed the package file. 

I'm not quite sure I understand what type failures could by caused by adding the typing files for `luxon`. If it's type mismatches that happened due to now knowing the types, I'm inclined to say that those extra errors are good to have.

I see some 30-40 usages of `luxon` and 6 for `moment`, maybe it's easier to standardize on `luxon`?

sylenien commented on frontend/app/player/MessageDistributor/managers/DOM/DOMManager.ts:
```diff
@@ -103,6 +103,7 @@ export default class DOMManager extends ListWalker<Message> {
         pNode.sheet && 
         pNode.sheet.cssRules &&
         pNode.sheet.cssRules.length > 0 &&
+        pNode.innerText &&
```
whats going on here

Aeolun replied:
I had a `cannot call method trim() on undefined` error, and I could find the thing that was undefined, so I added undefined checks to all instances where trim was used.

sylenien commented on frontend/app/styles/colors-autogen.css:
```diff
@@ -35,6 +35,7 @@
 .fill-light-blue-bg { fill: $light-blue-bg }
 .fill-white { fill: $white }
 .fill-borderColor { fill: $borderColor }
+.fill-figmaColors { fill: $figmaColors }
```
not sure what's happened here since 'figmacolors' is an object inside theme/colors

Aeolun replied:
Rebase issue I think. Removed all.

sylenien reviewed (changes requested):
Mostly small fixes are left, mainly for the styling to keep the current theme colors etc

sylenien commented:
> (I'd fix them if I knew where they were, but there's 700 other type errors cluttering up my results 😛 ).

Yeah it's a case for me too, I try to fix little by little when I'm doing something.

> I see some 30-40 usages of `luxon` and 6 for `moment`, maybe it's easier to standardize on `luxon`?

I'm not sure about the decision on luxon vs moment, maybe it's worth to ask @shekarsiri

shekarsiri commented on frontend/app/components/Session/IOSPlayer/Network.js:
```diff
@@ -10,82 +10,85 @@ import TimeTable from 'Components/Session_/TimeTable';
-	  width: 80,
-	  render: (r) => `${r.duration}ms`,
-	}
+  {
```
@sylenien yes, we are not using it currently.

Aeolun commented on frontend/app/components/Session_/BottomBlock/bottomBlock.module.css:
```diff
@@ -4,6 +4,4 @@
   /* padding-right: 10px; */
   /* border: solid thin $gray-light; */
   height: 300px;
```
I don't have a strong opinion on look, so if I modified something somewhere, it's very likely because it's inconsistent (this padding here is re-applied elsewhere, leading to double padding in the information header). 

The border removal is accidental. I rebased on a 7 month old dev branch first, where the border didn't exist I think.

Aeolun commented on frontend/app/components/Session_/GraphQL/GraphQL.js:
```diff
@@ -1,87 +1,136 @@
 import React from 'react';
-import { NoContent, Input, SlideModal, CloseButton } from 'UI';
+import { NoContent, Input, SlideModal, CloseButton, Popup, Button } from 'UI';
```
Correct, guess it was auto imported at some point.

Aeolun commented on frontend/app/components/Session_/StackEvents/StackEvents.js:
```diff
@@ -1,85 +1,176 @@
-@connectPlayer(state => ({
+@connectPlayer((state) => ({
   stackEvents: state.stackList,
+  stackEventsNow: state.stackListNow,
```
I could, but it wouldn't change anything since it'd still filter (and then render) any time something changed in the store.

I think the selector solution is probably still most optimal, but it's a bit beyond the scope of this PR.

Aeolun commented on frontend/app/components/Session_/StackEvents/UserEvent/userEvent.module.css:
```diff
@@ -2,9 +2,6 @@
   background-color: rgba(0, 118, 255, 0.05);
-  font-family: 'Menlo', 'monaco', 'consolas', monospace;
-  padding: 8px 10px;
-  margin: 3px 0;
```
I restored the font. It looked really crazy because it was applied to the entire item, so I've switched it to the infoWrapper (so it only applies to what it applied to before I added a timestamp). 

The padding used for the items in console, exceptions and events (all the non table information) was different for each. I'm happy to use whatever, but I think they should all use the same padding, not something different everywhere.

Aeolun commented on frontend/app/components/Session_/Console/console.module.css:
```diff
@@ -18,11 +18,18 @@
 .activeRow {
-  background-color: $teal !important;
-  color: white !important;
+  background-color: rgba(54, 108, 217, 0.1) !important;
```
Modified them to `$teal-light` everywhere.

Aeolun commented on frontend/app/components/Session_/TimeTable/TimeTable.tsx:
```diff
@@ -11,31 +12,33 @@ import stl from './timeTable.module.css';
+  dataKey?: string;
+  render?: (row: any) => void
+  referenceLines?: Array<Line>;
+  style?: Object;
```
Made it `React.CSSProperties`, better for things that can be a number (plus easy autocomplete).

Aeolun commented on frontend/app/components/Session_/TimeTable/TimeTable.tsx:
```diff
@@ -11,31 +12,33 @@ import stl from './timeTable.module.css';
 };
 
-type Row = Timed & Durationed & CanBeRed;
+type Row = Timed & Durationed & CanBeRed & { [key: string]: any, key: string };
```
That works

Aeolun commented on frontend/app/components/Dashboard/Widgets/common/widgetHOC.js:
```diff
@@ -8,101 +8,142 @@ import { WIDGET_MAP } from 'Types/dashboard';
-	@connect((state, props) => {
-		const compare = props && props.compare;
-		const key = compare ? '_' + widgetKey : widgetKey;
+export default (widgetKey, panelProps = {}, wrapped = true, allowedFilters = []) =>
```
The reformatting prevents a type/compilation error on my side, I'm not sure how it works on yours.

sylenien replied:
I see :)

Aeolun commented:
@sylenien  I've adjusted most of the things you asked, and responded to the comments where I didn't (or not fully) did what you requested.

I'm not sure what you want to do with the files that are unused? I can revert them, or we can leave the prettier formatting for what it is (it doesn't look like there are any functional changes).

sylenien reviewed (approved):
I think we can leave the unused files as is, maybe one day we will make some of them used again and this refactoring will help.
I'm only waiting for @shekarsiri green light here

PandaCodes commented on frontend/app/player/MessageDistributor/messages/urlResolve.ts:
```diff
@@ -29,7 +29,7 @@ function cssUrlsIndex(css: string): Array<[number, number]> {
 }
 function unquote(str: string): [string, string] {
-  str = str.trim();
+  str = str ? str.trim() : '';
```
🤔

PandaCodes replied:
This is not that trim for sure

Aeolun replied:
Never trust the types 😅

sylenien commented:
Hi again, could you please resolve the conflicts? Then we will merge it finally :)

Aeolun commented:
@sylenien @shekarsiri Rebased, feel free to merge any time 🙂

sylenien commented:
Done, thank you for your contribution.
